cmd/ci-operator-checkconfig/main.go

```diff
@@ -113,6 +118,15 @@ func (o *options) loadResolver(path string) error {
 	return nil
 }
 
+func (o *options) createCIOpConfigAgent() error {
+	ciOPConfigAgent, err := agents.NewConfigAgent(o.ConfigDir)
```

Comment:
We're now loading configuration files twice.  This is by far the largest factor in execution time, which can be seen in the doubling caused by these changes.

Can we use `GetAll` in `validate` to avoid this?

https://github.com/openshift/ci-tools/blob/d79b51ab614ceb9c1a4f487b91d45f62b7457b9a/pkg/load/agents/configAgent.go#L161

Reply:
Good point. My bash script wont suffer from this, for sure. :) 
Let me give it a try today.

Reply:
`GetAll` gives you the configurations once they are loaded.  The agent loads them when it is created, so any errors will already have been reported.

`GetAll` would replace the `produce` part of the pipeline: you'd remove `OperateOnCIOperatorConfigDir` (removing the duplicated loading of the files) and instead send the values returned by the agent to `inputCh`.

This will remove the parallelism we have between the producer and the mappers, but it shouldn't be too bad (we can measure to be certain).

Reply:
Let me give another try tomorrow.